Makefile

```diff
@@ -0,0 +1,346 @@
+
+# Disable implicit rules
+MAKEFLAGS += --no-builtin-rules
+
+ifndef OS
+$(error "OS must be set to 'lnx', 'win', or 'osx'")
+endif
+
+# The extra tag to put on ./internal/temp and qbx.o when multiple instances are involved
+# This is blank for the 'normal' files
+TEMP_ID ?=
+
+# Extra flags go at the beginning of the library list
+#
+# This is important for libraries, since they could potentially be referencing
+# things from our dependencies
+CXXFLAGS += $(CXXFLAGS_EXTRA)
```

Comment:
Note that related to #53 the new Makefile approach does not strip the executables, it's missing that flag. I should add that back as not doing that adds a significant size to the executables, but I'm thinking I may just try to do #53 as part of this change.